### PR TITLE
[ADD] IP added under the config file default_settings.yaml are now ab…

### PR DESCRIPTION
**Pull request by Jurdn**
…le to access site even under maintenance.

| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | yes 
| BC breaks?    | no
| Fixed tickets | #23 

- Added new feature on MaintenanceBundle to let authorized IP (in default_settings.yml) to connect to the website even if it's in maintenance mode. 
        - >You can now add a maintenance policy to users to let them access the maintenance admin panel.

Fixed crash when user tries to access admin routes without the /admin prefix by checking user rights. It now redirects to authentication if user is not logged in instead of just crashing.

image

### CODE REVIEW

**Review by okante — COMMENTED** [review]
Can you please make:
services lazy (add lazy: true) to services definitions in service.yml
Use 
```php
public static function getSubscribedEvents(): array
    {
        return [
            KernelEvents::REQUEST => [
                ['onKernelRequest', 10],
            ],
        ];
    }
``` 
instead of KernelEvents::RESPONSE,
doing that you should also modif `onKernelRequest` parameter to `Symfony\Component\HttpKernel\Event\RequestEvent;` according to the documentation

File: components/MaintenanceBundle/bundle/Controller/MaintenanceController.php [hunk 1]
```diff
@@ -47,21 +49,25 @@ public function __construct(FlashBagInterface $flashBag, FileHelper $fileHelper)
      * @Route("/", name="novamaintenance_index")
      * @Template("@ezdesign/maintenance/index.html.twig")
      */
-    public function indexAction(): array
+    public function indexAction(PermissionResolver $permissionResolver): Response
```
Comment (okante) [inline components/MaintenanceBundle/bundle/Controller/MaintenanceController.php:52]: remove permission resolver from hier

File: components/MaintenanceBundle/bundle/Controller/MaintenanceController.php [hunk 2]
```diff
@@ -47,21 +49,25 @@ public function __construct(FlashBagInterface $flashBag, FileHelper $fileHelper)
...
     /**
      * @Route("/manage/{maintenanceSiteaccess}", name="novamaintenance_manage")
-     * @Template("@ezdesign/maintenance/manage.html.twig")
      */
-    public function manageAction(string $maintenanceSiteaccess, Request $request): array
-    {
+    public function manageAction(
+        string $maintenanceSiteaccess,
+        Request $request,
+        PermissionResolver $permissionResolver
```
Comment (okante) [inline components/MaintenanceBundle/bundle/Controller/MaintenanceController.php:67]: remove unused permission Resolver

**Review by okante — CHANGES_REQUESTED** [review]

File: components/MaintenanceBundle/bundle/Controller/MaintenanceController.php [hunk 3]
```diff
@@ -47,21 +48,22 @@ public function __construct(FlashBagInterface $flashBag, FileHelper $fileHelper)
      * @Route("/", name="novamaintenance_index")
      * @Template("@ezdesign/maintenance/index.html.twig")
```
Comment (florianalexandre) [inline components/MaintenanceBundle/bundle/Controller/MaintenanceController.php:49]: not used anymore right ?

File: components/MaintenanceBundle/bundle/Resources/translations/forms.en.yml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+role.policy.novamaintenance: 'Maintain'
```
Comment (florianalexandre) [inline components/MaintenanceBundle/bundle/Resources/translations/forms.en.yml:1]: Maintenance

**Review by florianalexandre — CHANGES_REQUESTED** [review]

File: components/MaintenanceBundle/bundle/Controller/MaintenanceController.php (hunk 3 above)
Comment (Plopix) [reply components/MaintenanceBundle/bundle/Controller/MaintenanceController.php:49]: => template annotation right.

File: components/MaintenanceBundle/bundle/NovaeZMaintenanceBundle.php [hunk 5]
```diff
@@ -15,11 +15,25 @@
...
 class NovaeZMaintenanceBundle extends Bundle
 {
+    public function build(ContainerBuilder $container): void
+    {
+        parent::build($container);
+
+        // ...
+
+        // Retrieve "ezpublish" container extension
+        $eZExtension = $container->getExtension('ezpublish');
```
Comment (Plopix) [inline components/MaintenanceBundle/bundle/NovaeZMaintenanceBundle.php:32]: Can we double check if this is still `ezpublish`

**Review by Plopix — APPROVED** [review]
Ok with the few changes ;)

File: components/MaintenanceBundle/bundle/Resources/translations/forms.en.yml (hunk 4 above)
Comment (okante) [reply components/MaintenanceBundle/bundle/Resources/translations/forms.en.yml]: Done!

**Review by florianalexandre — APPROVED** [review]

**Pull request merged**